R/PythonPkgWrapperUtils.R

```diff
@@ -0,0 +1,807 @@
+          # inserting a value into a list at best is a no-op, at worst removes an existing value
+          # to get the desired insertion we must wrap it in a list
+          args[length(args) + 1] <- list(NULL)
+        } else {
+          args[[length(args) + 1]] <- values[[i]]
+        }
+      } else {
+        # It's a keyword argument.  All subsequent arguments must also be keyword arg's
+        positionalArgument <- FALSE
+        # a repeated value will overwite an earlier one
+        if (is.null(values[[i]])) {
+          # inserting a value into a list at best is a no-op, at worst removes an existing value
+          # to get the desired insertion we must wrap it in a list
+          kwargs[valuenames[[i]]] <- list(NULL)
+        } else {
+          kwargs[[valuenames[[i]]]] <- values[[i]]
+        }
+      }
+    }
+  }
+  list(args = args, kwargs = kwargs)
+}
+
+# The purpose of this function is to remove the Python stack trace from an error message
+#  generated when calling Python from R. This makes the command line response more readable
+#  when an error occurs. To support debugging the stack trace truncation can be overridden
+#  by setting the global option 'verbose' to TRUE.
+#
+# @param callable the function to be called
+# @param args the arguments to be passed to the function 'callable'
+# @return the result of calling the given function with the given arguments
+cleanUpStackTrace <- function(callable, args) {
+  conn <- textConnection("outputCapture", open = "w", local = TRUE)
+  sink(conn)
+  tryCatch({
+    result <- do.call(callable, args)
+    sink()
+    close(conn)
+    cat(paste(outputCapture, collapse = ""))
+    result
+  },
+  error = function(e) {
+    sink()
+    close(conn)
+    errorToReport <- paste(c(outputCapture, e$message), collapse = "\n")
+    if (!getOption("verbose")) {
+      # extract the error message
+      splitArray <- strsplit(errorToReport,
+        "exception-message-boundary",
+        fixed = TRUE
+      )[[1]]
+      if (length(splitArray) >= 2) errorToReport <- splitArray[2]
+    }
+    stop(errorToReport)
+  }
+  )
+}
+
+#' @title Generate R wrappers for Python classes and functions
+#' @description This function generate R wrappers for Python classes and functions
+#'  in the given Python module
+#'
+#' @param pyPkg the Python package name
+#' @param module the Python module
+#' @param setGenericCallback the callback to setGeneric defined in the target R package
+#' @param modifyFunctions optional function to modify the returned functions
+#' @param modifyClasses optional function to modify the returned classes
+#' @param functionPrefix optional text to add to the name of the functions
+#' @param pyObjectName optional singleton object in python
+#' @param transformReturnObject optional function to change returned values in R
+#' @param replaceParam optional function to replace a function param
+#' @details generateRdFiles and generateRWrappers should be called with similar
```

Comment:
the details section is a likely place to say more about param's like `modifyFunctions()`.